code/espurna/sensors/BME680Sensor.h

```diff
@@ -0,0 +1,242 @@
+// -----------------------------------------------------------------------------
+// BME680 Sensor over I2C
+// Copyright (C) 2017-2019 by Xose Pérez <xose dot perez at gmail dot com>
+// -----------------------------------------------------------------------------
+
+#if SENSOR_SUPPORT && BME680_SUPPORT
+
+#pragma once
+
+#include "Arduino.h"
+#include "I2CSensor.h"
+#include "bsec.h"
+
+// Configuration profile `generic_33v_3s_4d`. The default is `generic_18v_300s_4d`.
+// In the future, this can be made dynamic.
+//
+// - 3.3V as supply voltage for BME680.
+// - 3 seconds as maximum time between bsec_sensor_control` calls.
+// - 4 days as the time considered for background calibration.
+const uint8_t bsec_config_iaq[454] = {
```

Comment:
I missed this part not being included from the library sources. Does it need some include path tweaks to have `#include <config/generic_33v_3s_4d.txt>`?

Reply:
Ah good catch, as usual. The original instructions of BSEC suggested copying over the config file to the right directory, so I thought that was too cumbersome - inlining would make it simpler. However, this can be safely be replaced by:

```cpp
const uint8_t bsec_config_iaq[] = {
  #include <config/generic_33v_3s_4d/bsec_iaq.txt>
};
```

Is there a way to make 'computed includes'? That way we could move this to config and have `generic_33v_3s_4d` be a configurable define option.

Reply:
examples do have it included, I was looking at those
https://github.com/BoschSensortec/BSEC-Arduino-library/blob/dcd55be0cab81807f7fe1164f6f1793fbc8cdeaf/examples/basic_config_state/basic_config_state.ino#L3-L15
```cpp
/* Configure the BSEC library with information about the sensor
    18v/33v = Voltage at Vdd. 1.8V or 3.3V
    3s/300s = BSEC operating mode, BSEC_SAMPLE_RATE_LP or BSEC_SAMPLE_RATE_ULP
    4d/28d = Operating age of the sensor in days
    generic_18v_3s_4d
    generic_18v_3s_28d
    generic_18v_300s_4d
    generic_18v_300s_28d
    generic_33v_3s_4d
    generic_33v_3s_28d
    generic_33v_300s_4d
    generic_33v_300s_28d
*/
```

One option is to do something like:
```cpp
#if BME680_BSEC_CONFIG == BME680_BSEC_CONFIG_GENERIC_33V_3S_4D
#include <config/generic_33v_3s_4d/bseq_iaq.txt>
#elif BME680_BSEC_CONFIG == ...
...
#endif
```